src/endpoint/s3/ops/s3_put_bucket_policy.js

```diff
@@ -10,21 +9,22 @@ const S3Error = require('../s3_errors').S3Error;
 async function put_bucket_policy(req) {
     let policy;
     try {
-        policy = JSON.parse(JSON.stringify(req.body).toLowerCase());
+        policy = JSON.parse(JSON.stringify(req.body));
     } catch (error) {
         console.error('put_bucket_policy: Invalid JSON provided', error);
         throw new S3Error(S3Error.InvalidArgument);
     }
-    // adapting bucket policy to our schema
-    for (const statement of policy.statement) {
-        if (statement.principal) {
-            if (statement.principal.aws) statement.principal = statement.principal.aws;
-            statement.principal = _.flatten([statement.principal]);
+    try {
+        await req.object_sdk.put_bucket_policy({ name: req.params.bucket, policy });
+    } catch (error) {
+        if (error.rpc_code === "INVALID_SCHEMA") {
+            const err = new S3Error(S3Error.MalformedPolicy);
+            err.message = "Policy was not well formed or did not validate against the published schema";
+            throw err;
+        } else {
```

Comment:
if the malformed condition is needed, you can remove the else and the first throw err and just throw err after the condition